modules/internal/ChapelArray.chpl

```diff
@@ -2659,6 +2723,14 @@ module ChapelArray {
   }
 
   proc =(ref a: domain, b: domain) {
+    if a.rank != b.rank then
+      compilerError("rank mismatch in domain assignment");
+    if a.idxType != b.idxType then
+      compilerError("index type mismatch in domain assignment");
+    if isRectangularDom(a) && isRectangularDom(b) then
+      if !a.stridable && b.stridable then
+        compilerError("stridable mismatch in domain assignment");
```

Comment:
Updating existing errors is up to you. Clearer error messages are always welcome and are not the top priority for me. I am fine if the new error message's phrasing differs from existing messages.